scripts/lint/check-complexity.sh

```
@@ -1,30 +1,65 @@
-    if ! head -n "${lno}" "${file}" | tail -n 6 | grep -q "pmccabe:skip-complexity\|clang-tidy:skip-complexity"; then
-      echo "${line}" | awk '$1 > 10 {print; exit 1}'
+  if ! command -v pmccabe >/dev/null; then
+    print_step "" "check-complexity" skip "pmccabe not installed"
+    exit 0
+  fi
+
+  print_header "checking complexity (threshold: $THRESHOLD)..."
+
+  # Step 1: get ALL complexities in one pass (fast)
+  local violations
+  violations=$(find src -name '*.cpp' | xargs pmccabe 2>/dev/null | awk -v t="$THRESHOLD" '$1 > t')
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)